web-components/src/components/menu-overlay/MenuOverlay.ts

```diff
@@ -115,14 +115,13 @@ export namespace MenuOverlay {
     connectedCallback() {
       super.connectedCallback();
       document.addEventListener("click", this.handleOutsideOverlayClick);
-      document.addEventListener("keydown", this.handleOutsideOverlayKeydown);
+      this.addEventListener("keydown", this.handleOutsideOverlayKeydown);
```

Comment:
Adding event listener in document is needed as it listens in document level to close the overlay. It means if any of the menu overlay is clicked , the other menu overlay will be closed.

Reply:
With this change we are targeting to close just the top overlay and not others, for every overlay to close we need to press escape

Reply:
What about the functionality impact at other places where it is used?

Reply:
removed this